crates/kornia-dnn/src/rtdetr.rs

```diff
@@ -0,0 +1,174 @@
+//! # RT-DETR
+//!
+//! This module contains the RT-DETR model.
+//!
+//! The RT-DETR model is a state-of-the-art object detection model.
+
+use std::path::PathBuf;
+
+use crate::error::DnnError;
+use crate::Detection;
+use kornia_core::{CpuAllocator, Tensor};
+use kornia_image::Image;
+use ort::{GraphOptimizationLevel, Session};
+
+/// Builder for the RT-DETR detector.
+///
+/// This struct provides a convenient way to configure and create an `RTDETRDetector` instance.
+pub struct RTDETRDetectorBuilder {
+    /// Path to the RT-DETR model file.
+    pub model_path: PathBuf,
+    /// Path to the ONNX Runtime dynamic library.
+    pub ort_dylib_path: PathBuf,
+    /// Number of threads to use for inference.
+    pub num_threads: usize,
+}
+
+impl RTDETRDetectorBuilder {
+    /// Creates a new `RTDETRDetectorBuilder` with default settings.
+    ///
+    /// # Arguments
+    ///
+    /// * `model_path` - Path to the RT-DETR model file.
+    /// * `ort_dylib_path` - Path to the ONNX Runtime dynamic library.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing the `RTDETRDetectorBuilder` if successful, or a `DnnError` if an error occurred.
+    pub fn new(model_path: PathBuf, ort_dylib_path: PathBuf) -> Result<Self, DnnError> {
+        Ok(Self {
+            model_path,
+            ort_dylib_path,
+            num_threads: 4,
+        })
+    }
+
+    /// Sets the number of threads to use for inference.
+    ///
+    /// # Arguments
+    ///
+    /// * `num_threads` - The number of threads to use.
+    ///
+    /// # Returns
+    ///
+    /// The updated `RTDETRDetectorBuilder` instance.
+    pub fn with_num_threads(mut self, num_threads: usize) -> Self {
+        self.num_threads = num_threads;
+        self
+    }
+
+    /// Builds and returns an `RTDETRDetector` instance.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing the `RTDETRDetector` if successful, or a `DnnError` if an error occurred.
+    pub fn build(self) -> Result<RTDETRDetector, DnnError> {
+        RTDETRDetector::new(self.model_path, self.ort_dylib_path, self.num_threads)
+    }
+}
+
+/// RT-DETR object detector.
+///
+/// This struct represents an instance of the RT-DETR object detection model.
+pub struct RTDETRDetector {
+    session: Session,
+}
+
+impl RTDETRDetector {
```

Comment:
@decahedron1 we got RTDETR working and seems pretty smooth. Woulk like to hear you opinion about this api or any considerations we should have for future models